docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx

```diff
@@ -7,39 +7,458 @@ description: Upgrade your Next.js Application from Version 14 to 15.
+
+### `cookies`
+
+#### Recommended Async Usage
+
+```tsx
+import { cookies } from 'next/headers'
+
+// Before
+const cookieStore = cookies()
+const token = cookieStore.get('token')
+
+// After
+const cookieStore = await cookies()
+const token = cookieStore.get('token')
+```
+
+#### Temporary Synchronous Usage
+
+```tsx filename="app/page.tsx" switcher
+import { cookies, type UnsafeUnwrappedCookies } from 'next/headers'
+
+// Before
+const cookieStore = cookies()
+const token = cookieStore.get('token')
+
+// After
+const cookieStore = cookies() as unknown as UnsafeUnwrappedCookies
+// will log a warning in dev
+const token = cookieStore.get('token')
+```
+
+```jsx filename="app/page.js" switcher
+import { cookies } from 'next/headers'
+
+// Before
+const cookieStore = cookies()
+const token = cookieStore.get('token')
+
+// After
+const cookieStore = cookies()
+// will log a warning in dev
+const token = cookieStore.get('token')
+```
+
+### `headers`
+
+#### Recommended Async Usage
+
+```tsx
+import { headers } from 'next/headers'
+
+// Before
+const headersList = headers()
+const userAgent = headersList.get('user-agent')
+
+// After
+const headersList = await headers()
+const userAgent = headersList.get('user-agent')
+```
+
+#### Temporary Synchronous Usage
+
+```tsx filename="app/page.tsx" switcher
+import { headers, type UnsafeUnwrappedHeaders } from 'next/headers'
+
+// Before
+const headersList = headers()
+const userAgent = headersList.get('user-agent')
+
+// After
+const headersList = headers() as unknown as UnsafeUnwrappedHeaders
+// will log a warning in dev
+const userAgent = headersList.get('user-agent')
+```
+
+```jsx filename="app/page.js" switcher
+import { headers } from 'next/headers'
+
+// Before
+const headersList = headers()
+const userAgent = headersList.get('user-agent')
+
+// After
+const headersList = headers()
+// will log a warning in dev
+const userAgent = headersList.get('user-agent')
+```
+
+### `draftMode`
+
+#### Recommended Async Usage
+
+```tsx
+import { draftMode } from 'next/headers'
+
+// Before
+const { isEnabled } = draftMode()
+
+// After
+const { isEnabled } = await draftMode()
+```
+
+#### Temporary Synchronous Usage
+
+```tsx filename="app/page.tsx" switcher
+import { draftMode, type UnsafeUnwrappedDraftMode } from 'next/headers'
+
+// Before
+const { isEnabled } = draftMode()
+
+// After
+// will log a warning in dev
+const { isEnabled } = draftMode() as unknown as UnsafeUnwrappedDraftMode
+```
+
+```jsx filename="app/page.js" switcher
+import { draftMode } from 'next/headers'
+
+// Before
+const { isEnabled } = draftMode()
+
+// After
+// will log a warning in dev
+const { isEnabled } = draftMode()
+```
+
+### `params` & `searchParams`
+
+#### Asynchronous Layout
+
+```tsx filename="app/layout.tsx" switcher
+// Before
+type Params = { slug: string }
+
+export function generateMetadata(props: { params: Params }) {
+  const params = props.params
```

Comment:
it looks bit weird here that the js code has no destructuring but types does, how about simplify it a bit? I feel users might want to see the code that similar to what they have, this might be not common in the real world. It took me 1sec to connect with it

```
export function generateMetadata({ params }: { params: Params }) {
  const { slug } = params
```